### PR TITLE
Enable HTTP trailer support

### PR DESCRIPTION
**Pull request by filakhtov**
Hey guys!

Currently playing with Laravel and Swoole and noticed that there is no support for HTTP Trailers.

Swoole supports trailer response headers, but Laravel does not (same as almost all other PHP frameworks). This presents a challenge, where we either need to extend Laravel's (Symfony's to be precise) HTTP response object to include trailers support or take a different strategy.

In this PR, the proposal is to take a different approach, i.e. rely on the `trailer` header to specify the list of all the headers that should be converted into trailing. Similar approach has been taken for [RoadRunner application server](https://github.com/spiral/roadrunner/blob/fea88ad23d8686ec6ddbeb48f5cfc7b2640c036e/service/http/response.go#L89-L107) and worked well, so the hope is that this is a reasonable solution for LaravelS as well. It is backwards compatible with Laravel's response and supports a missing feature properly.

Open to any feedback and suggestions! Thanks in advance.

### CODE REVIEW

File: src/Swoole/Response.php [hunk 1]
```diff
@@ -30,17 +30,49 @@ public function sendStatusCode()
...
+        return $this->laravelResponse->headers->allPreserveCase();
+    }
+
     public function sendHeaders()
     {
-        $headers = method_exists($this->laravelResponse->headers, 'allPreserveCaseWithoutCookies') ?
-            $this->laravelResponse->headers->allPreserveCaseWithoutCookies() : $this->laravelResponse->headers->allPreserveCase();
+        $headers = $this->getHeaders();
+
+        $trailers = $headers["trailer"] ?? [];
```
Comment (hhxsv5) [inline src/Swoole/Response.php:46]: Same as above

File: src/Swoole/Response.php [hunk 2]
```diff
@@ -30,17 +30,49 @@ public function sendStatusCode()
...
                 $this->swooleResponse->header($name, $value);
             }
         }
     }
 
+    public function sendTrailers()
+    {
+        $headers = $this->getHeaders();
+
+        $trailers = $headers["trailer"] ?? [];
```
Comment (hhxsv5) [inline src/Swoole/Response.php:63]: Same as above

**Review by hhxsv5 — CHANGES_REQUESTED** [review]
Thanks for your contribution. 
If it is compatible with PHP 5.x, so much better

**Comment by filakhtov** [comment]
Hey @hhxsv5, thanks a lot for the review and your feedback. I have made a requested changes and it is ready for another round, whenever you get a chance 🙏 

I have a quick question for you as well that I am curious about: what is the motivation to keep the compatibility with PHP <7.2? According to the [supported versions page](https://www.php.net/supported-versions.php) from the official PHP website, everything older than 7.2 is EOL and there are no active support or security patches, and even 7.2 is going to EOL in 20 days.

**Comment by hhxsv5** [comment]
> Hey @hhxsv5, thanks a lot for the review and your feedback. I have made a requested changes and it is ready for another round, whenever you get a chance 🙏
> 
> I have a quick question for you as well that I am curious about: what is the motivation to keep the compatibility with PHP <7.2? According to the [supported versions page](https://www.php.net/supported-versions.php) from the official PHP website, everything older than 7.2 is EOL and there are no active support or security patches, and even 7.2 is going to EOL in 20 days.

Just to make Swoole available for more old projects 😁.

**Pull request merged**